Mathlib/Data/ListM/Basic.lean

```diff
@@ -111,8 +111,8 @@ instance [Monad m] [MonadLiftT m n] : ForIn n (ListM m α) α where
   forIn := ListM.forIn
 
 /-- Construct a `ListM` recursively. -/
-partial def fix [Monad m] [Alternative m] (f : α → m α) (x : α) : ListM m α :=
-  cons' x <| squash <| (fix f <$> f x) <|> pure nil
+partial def fix [Monad m] (f : α → m α) (x : α) : ListM m α :=
+  cons' x <| squash <| fix f <$> f x
```

Comment:
The rest of the API now just fails when moadic computations fail; this makes this consistent.